submissions/salsita/src/reducers/sithLoadingReducer.js

```diff
@@ -0,0 +1,75 @@
+import { SithRecord } from '../constants/records';
+
+// This simply initializes the application state and triggers loading of first Sith
+export const applicationMounting = reduction => reduction.withMutations(mutableReduction => {
+  mutableReduction
+    .setIn(['appState', 'disabledUp', true])
+    .setIn(['appState', 'disabledDown', true])
+    .setIn(['appState', 'scrollingDown'], true)
+    .setIn(['appState', 'sithsSpotted'], false)
+    .setIn(['appState', 'siths'], new Range(0, Configuration.SITH_SLOTS).map(() => new SithRecord()).toList())
+    .setIn(['appState', 'siths', 0, 'id'], Configuration.INITIAL_SITH_TO_LOAD)
+    .setIn(['appState', 'siths', 0, 'state'], SithStates.LOADING)
+    .setIn(['appState', 'planet'], null)
+    .update('effects', effects => effects
+      .push(buildMessage(Effects.API_LOAD_SITH, {
+        sithId: Configuration.INITIAL_SITH_TO_LOAD,
+        scrollingDown: true
+      }))
+      .push(buildMessage(Effects.API_CONNECT_WS)));
+});
+
+export const sithLoaded = (reduction, payload) => {
+  const loadedSithIndex = reduction.getIn(['appState', 'siths']).findIndex(sith => sith.get('id') === payload.sith.id);
+  const scrollingDown = payload.scrollingDown;
+  const apprentice = payload.sith.apprentice.id;
+  const master = payload.sith.master.id;
+  const sithName = payload.sith.name;
+
+  return reduction.withMutations(mutableReduction => {
+    // Let's store the newly loaded Sith
+    mutableReduction
+      .updateIn(['appState', 'siths', loadedSithIndex], sith => sith
+        .set('state', SithStates.LOADED)
+        .set('correlationId', null)
+        .set('apprentice', apprentice)
+        .set('master', master)
+        .set('homeworldId', payload.sith.homeworld.id)
+        .set('homeworldName', payload.sith.homeworld.name)
+        .set('name', sithName));
+
+    if (scrollingDown && apprentice && loadedSithIndex < (Configuration.SITH_SLOTS - 1)) {
+      const nextSithIndex = loadedSithIndex + 1;
+
+      mutableReduction
+        .setIn(['appState', 'siths', nextSithIndex, 'id'], apprentice)
+        .setIn(['appState', 'siths', nextSithIndex, 'state'], SithStates.LOADING)
+        .update('effects', effects => effects.push(buildMessage(Effects.API_LOAD_SITH, {
+          sithId: apprentice,
+          scrollingDown: true
+        })));
+    } else if (!scrollingDown && master && loadedSithIndex > 0) {
+      const prevSithIndex = loadedSithIndex - 1;
+
+      mutableReduction
+        .setIn(['appState', 'siths', prevSithIndex, 'id'], master)
+        .setIn(['appState', 'siths', prevSithIndex, 'state'], SithStates.LOADING)
+        .update('effects', effects => effects.push(buildMessage(Effects.API_LOAD_SITH, {
+          sithId: master,
+          scrollingDown: false
+        })));
+    }
+  });
+};
+
+// Let's store the correlation ID of the request so that we can cancel it later on if necessary
+export const sithLoadingStarted = (reduction, payload) => {
```

Comment:
Total nitpick but I would put this before `sithLoaded`. It makes the code easier to understand to put them in the order that they are triggered.